scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -32,23 +32,43 @@
   ; suppress otherwise identical states so we don't spam.
   (= old-state new-state))
 
-(defn remove-finalization-if-set-and-delete
-  [api-client expected-state-dict pod]
-  (api/remove-finalization-if-set-and-delete api-client pod)
-  expected-state-dict)
+(defn delete-task
+  "Kill task is the same as deleting a task. I semantically distinguish them. Delete is used for completed tasks that
+  we're done with. Kill is used for possibly running tasks we want to kill so that they fail. Returns a new expected state
+  dict of nil."
+  [api-client pod]
+  (api/kill-task api-client pod)
+  nil) ; Return new expected state dict of nil.
 
 (defn kill-task
+  "Kill task is the same as deleting a task. I semantically distinguish them. Delete is used for completed tasks that
+  we're done with. Kill is used for possibly running tasks we want to kill so that they fail."
   [api-client expected-state-dict pod]
   (api/kill-task api-client pod)
   expected-state-dict)
 
+(defn weird
+  "Weird. This pod is in a weird state. Log its weird state."
+  [expected-state-dict pod]
```

Comment:
Why is this called `expected-state-dict`? I think it should be `current-state-dict`.

Reply:
I wish you would call it `current-state` as that is what it is, the current state of the task/pod.

Reply:
This is why I don't want to use current state.  You're proposing using 'current-state' instead of 'expected-state-dict' because it is the current state of the task/pod. The current state of the pod is stored in 'existing-state-dict' --- the other dictionary. I can't rename both dictionaries to current-state.

Reply:
Lets call the functions cook-expected-state   k8s-actual-state  
Rename the state names from expected/ cook-expected-state/*